rholang/src/main/scala/coop/rchain/rholang/interpreter/Runtime.scala

```diff
@@ -425,27 +419,30 @@ object Runtime {
         .getBytes()
     )
 
-    val cost0: _cost[F] =
-      loggingCost(CostAccounting.unsafe[F](Cost.UNSAFE_MAX), noOpCostLog)
-    val cost1: _cost[F] =
-      loggingCost(CostAccounting.unsafe[F](Cost.UNSAFE_MAX), noOpCostLog)
-
     for {
-      spaceResult <- space.produce(
-                      Registry.registryRoot,
-                      ListParWithRandom(Seq(Registry.emptyMap), rand),
-                      false,
-                      0
-                    )(matchListPar(F, cost0))
-      replayResult <- replaySpace.produce(
-                       Registry.registryRoot,
-                       ListParWithRandom(Seq(Registry.emptyMap), rand),
-                       false,
-                       0
-                     )(matchListPar(F, cost1))
-      _ <- spaceResult match {
+      costAlg <- CostAccounting.of[F](Cost(0))
+      resultsF = {
+        implicit val cost0: _cost[F] = loggingCost(costAlg, noOpCostLog)
+        (noCharge {
+          space.produce(
+            Registry.registryRoot,
+            ListParWithRandom(Seq(Registry.emptyMap), rand),
+            false,
+            0
+          )(matchListPar)
+        }, noCharge {
+          replaySpace.produce(
+            Registry.registryRoot,
+            ListParWithRandom(Seq(Registry.emptyMap), rand),
+            false,
+            0
+          )(matchListPar)
+        })
+      }
```

Comment:
maybe we could have 
```scala
_ <- noCharge { costAlg =>
  implicit val cost0: _cost[F] = loggingCost(costAlg, noOpCostLog)
  space.produce(...)
}
_ <- noCharge { costAlg =>
  implicit val cost0: _cost[F] = loggingCost(costAlg, noOpCostLog)
  replaySpace.produce(...)
}
```
here?

Since we don't charge anyway, not-sharing the cost account between the not-charging parts makes sense - especially given the boilerplate spared...

Reply:
I'm probably going to just `Cost.UNSAFE_MAX` for now and write `noCharge` into a subsequent PR 👍